### PR TITLE
15241 Switch

### PR DESCRIPTION
**Pull request by dhochbaum-dcp**
This adds the Switch component to our Storybook, as designed in [Figma](https://www.figma.com/file/i6Vvsi3UvR32mB8w8Qs80P/Design-System?node-id=258%3A7787&mode=dev).

This completes task AB#15241 and user story AB#15161.

### CODE REVIEW

File: src/theme/components/switch.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  thumb: baseStyleThumb,
+}));
+
+const sizes = {
+  sm: definePartsStyle({
+    container: { width: "28px", height: "16px" },
+    track: {
+      width: "12px",
+      height: "12px",
+      padding: "2px 14px 2px 2px",
```
Comment (TylerMatteo) [inline src/theme/components/switch.ts:43]: Instead of adding padding to the track to move the "thumb", let's see if we can use `translateX` and transitions on the thumb itself like [Chakra's default theme](https://github.com/chakra-ui/chakra-ui/blob/main/packages/components/theme/src/components/switch.ts#L57C12-L57C12). I realize that code is a little hard to follow, so happy to pair on this.

File: src/theme/components/switch.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      _checked: {
+        padding: "2px 2px 2px 14px",
+      },
+    },
+    thumb: { width: "12px", height: "12px" },
```
Comment (TylerMatteo) [inline src/theme/components/switch.ts:48]: Width and height should use `size` tokens. Unfortunately Figma can't show us things like height, margin, and padding in tokens so we'll have to convert from px or rems to tokens in our heads. For instance, the token for `"12px"` is `3`.

**Review by TylerMatteo — CHANGES_REQUESTED** [review]
Really just two things to address here:
* Using references to tokens wherever possible
* Using translate and transitions for moving the thumb
Otherwise looking good.

File: src/theme/components/switch.ts (hunk 1 above)
Comment (dhochbaum-dcp) [reply src/theme/components/switch.ts]: Do we need to use translateX?  The thumb appears to move appropriately due to the track's _checked { justifyContent: "flex-end", }

Comment (TylerMatteo) [reply src/theme/components/switch.ts]: using translate just gives it the animation that you [see in the docs](https://chakra-ui.com/docs/components/switch/usage). I pushed a commit to resolve that. Merging now

**Review by TylerMatteo — DISMISSED** [review]

**Review by TylerMatteo — APPROVED** [review]
